src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -86,5 +113,260 @@ private protected IReadOnlyList<PackageReference> GetTransitivePackageReferences
+
+                foreach (PackageDependency dep in node.Dependencies.ToList()) // Casting to list to prevent backing allocations
+                {
+                    // Create PackageReference object as a data-model based on dependency
+                    var pkgChild = new PackageReference(
+                        identity: new PackageIdentity(dep.Id, dep.VersionRange.MinVersion),
+                        targetFramework: fxRidEntry.Framework,
+                        userInstalled: false,
+                        developmentDependency: false,
+                        requireReinstallation: false,
+                        allowedVersions: dep.VersionRange);
+
+                    if (!visited.Contains(pkgChild.PackageIdentity))
+                    {
+                        MarkTransitiveOrigin(top, pkgChild, graph, visited, fxRidEntry, token);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates a cache key for Transitive Originas cache
+        /// </summary>
+        /// <param name="transitivePackage"></param>
+        /// <returns>A string with given key</returns>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal PackageIdentity GetTransitiveCacheKey(PackageIdentity transitivePackage)
+        {
+            return transitivePackage;
+        }
+
+        /// <summary>
+        /// Obtains cached entry for a given transitive package
+        /// </summary>
+        /// <param name="transitivePackage">Identity of transitive package</param>
+        /// <returns>A <see cref="TransitiveEntry"/> object, or <c>null</c> if not found</returns>
+        /// <seealso cref="ClearCachedTransitiveOrigins"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal TransitiveEntry GetCachedTransitiveOrigin(PackageIdentity transitivePackage)
+        {
+            var key = GetTransitiveCacheKey(transitivePackage);
+
+            if (TransitiveOriginsCache.ContainsKey(key))
+            {
+                return TransitiveOriginsCache[key];
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Replaces cached entry for a given transitive package with a <see cref="TransitiveEntry"/>
+        /// </summary>
+        /// <param name="transitivePackage">Identity of transitive package</param>
+        /// <param name="origins">Packages identified as package origins</param>
+        /// <seealso cref="ClearCachedTransitiveOrigins"/>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        internal void SetCachedTransitiveOrigin(PackageIdentity transitivePackage, TransitiveEntry origins)
+        {
+            var key = GetTransitiveCacheKey(transitivePackage);
+            TransitiveOriginsCache[key] = origins;
```

Comment:
I'm not sure there's too much value of abstracting the usage of the cache. 

In some places the caches are used directly, but in others they are abstracted into 1/2 line methods. 

It'll be easier to read/reduce the line count if you just use replace it with one line since GetTransitiveCacheKey will likely go away.

Reply:
Removed cache methods and replaced with comment and one-liners.